InscryptionAPI/Regions/RegionManager.cs

```diff
@@ -188,4 +188,89 @@ public static RegionData GetRandomRegionFromTier(int tier)
         }
         return valid[SeededRandom.Range(0, valid.Count, randomseed)];
     }
+
+    [HarmonyPatch(typeof(EncounterBuilder), nameof(EncounterBuilder.BuildTerrainCondition))]
+    [HarmonyPrefix]
+    private static bool ApplyTerrainCustomization(ref EncounterData.StartCondition __result, ref bool reachTerrainOnPlayerSide, int randomSeed)
+    {
+        var customregion = NewRegions.ToList().Find(x => x.Region == RunState.CurrentMapRegion);
+        if(customregion != null)
+        {
+            reachTerrainOnPlayerSide &= !customregion.DoNotForceReachTerrain;
+            __result = new EncounterData.StartCondition();
+            int num = SeededRandom.Range(customregion.MinTerrain, customregion.MaxTerrain, randomSeed++);
```

Comment:
I realize that this is a copy/paste job from DNSpy, but if this is going to be code that needs to be supported in the future, can you go ahead and give the num/flag variables meaningful names, to aid in future debugging?

Otherwise this looks good.

Reply:
done